### PR TITLE
test: test improvements and tsconfig consolidation

### PR DESCRIPTION
**Pull request by AndyTWF**
### Context

- The root tsconfig.json currently is the one we use to build the core SDK. This breaks IDEs in things like tests because they often expect the root tsconfig.json to cover all files you have in the project.

### Description

- Adds some extra tests in places where they are lacking (non-exhaustive).
- Changes the root tsconfig.json to cover all files and tests, so that code like our custom vitest assertions show up properly in IDEs.
- Creates a core/ level tsconfig and vite.config that we now use to build the core library.
- For now I've moved the `utils` folder into `src/core` to make this work, fully anticipating that it'll get removed entirely very shortly.

### Checklist

* [x] QA'd by the author.
* [x] Unit tests created (if applicable).
* [x] Integration tests created (if applicable).
* [x] Follow coding style guidelines found [here](https://github.com/ably/engineering/tree/main/best-practices).
* [x] TypeDoc updated (if applicable).
* [x] (Optional) Update documentation for new features.
* [x] Browser tests created (if applicable).
* [x] In repo demo app updated (if applicable).

### Testing Instructions (Optional)

N/A

### CODE REVIEW

File: src/core/room-lifecycle-manager.ts [hunk 1]
```diff
@@ -570,11 +570,13 @@ export class RoomLifecycleManager {
             break;
           }
           default: {
-            this._logger.error(`Unexpected channel state ${feature.channel.state}`);
-            throw new Ably.ErrorInfo(
+            this._logger.error(`Unexpected channel state`, { state: feature.channel.state });
```
[the remark this answers is not in the document]
Comment (AndyTWF) [reply src/core/room-lifecycle-manager.ts:573]: it'll just end up in ably-js's error logger too, so avoiding duplication :)

File: test/core/chat-api.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import * as Ably from 'ably';
+import { describe, expect, it, vi } from 'vitest';
+
+import { ChatApi } from '../../src/core/chat-api.ts';
+import { makeTestLogger } from '../helper/logger.ts';
+
+vi.mock('ably');
+
+describe('config', () => {
+  it('throws errors if Realtime returns ErrorInfo on non-paginated request', async () => {
```
Comment (splindsay-92) [inline test/core/chat-api.test.ts:10]:
```suggestion
  it('throws an error if Realtime returns ErrorInfo on non-paginated request', async () => {
```

File: src/core/room-lifecycle-manager.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (splindsay-92) [reply src/core/room-lifecycle-manager.ts:573]: Makes sense to me, thanks for explaining :)

**Review by splindsay-92 — APPROVED** [review]
Looks like just a flakey test holding this back, but this has been logged already, so I will approve.

**Pull request merged**